src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -1923,7 +1922,7 @@ void ObjectMonitor::wait(jlong millis, bool interruptible, TRAPS) {
     guarantee(node.TState != ObjectWaiter::TS_WAIT, "invariant");
     OrderAccess::loadload();
     if (has_successor(current)) clear_successor();
-    WasNotified = node._notified;
+    WasNotified = node.TState == ObjectWaiter::TS_ENTER;
```

Comment:
We don't really need the `WasNotified` local any more (which saves me from asking you to rename it `was_notified`.)